ext/pdo/tests/gh8626.phpt

```diff
@@ -0,0 +1,54 @@
+--TEST--
+GH-8626: PDOStatement->execute() failed, then execute successfully, errorInfo() information is incorrect
+--SKIPIF--
+<?php
+if (!extension_loaded('pdo') || !extension_loaded('pdo_mysql')) die('skip');
+$dir = getenv('REDIR_TEST_DIR');
+if (false == $dir) die('skip no driver');
+if (!str_starts_with(getenv('PDOTEST_DSN'), 'mysql')) die('skip non-mysql drivers');
```

Comment:
If the test is only meant for PDO_MySQL, there is no point in putting it into ext/pdo/tests and making it a redirect test. Instead, the test should be put into ext/pdo_mysql/tests. However, since there is a behavioral change for all PDO drivers, shouldn't we test it for all drivers?

Reply:
I'm not very familiar with other databases, so I had to use MySQL to test it.
And, I'm not quite sure why the I386 test failed.

Reply:
I think the problem the usage of `PHP_INT_MIN` to trigger a failure. `PHP_INT_MIN` is `-2147483648` on 32bit architectures, and that wouldn't be problem for the database. Can't we use something else to trigger a failure? I'm thinking about trying to insert a non numeric string into an INTEGER column, or such. That should work for other drivers as well.

Reply:
I've added mysql, pgsql, sqlite tests.
Other databases I am really not familiar with and cannot test.